dockerfiles/init/manifests/codenvy.env

```diff
@@ -86,17 +86,121 @@
 #CODENVY_SERVER_XMX=2048
 
 
-
 ########################################################################################
 #####                                                                              #####
-#####                                   DOCKER                                     #####
+#####                                 NETWORKING                                   #####
 #####                                                                              #####
-# IP Address
-#     The IP address of the Docker daemon that is running on your host. We do a 
-#     self-discovery to set this. You can combine this with values of DOCKER_HOST to 
-#     change communications from socket to TCP.
+#
+# Affects how the Codenvy server connects to the Docker Daemon, how the Che server and
+# workspaces establish connections to one another, and also how remote browser clients
+# should discover and connect to workspaces.
+#
+# Codenvy goes through a progression algorithm to establish the protocol, IP address and
+# port to establish communications with it is booting or starting a workspace.
+#
+# Browser --> Codenvy Server
+#    1. Default is 'http://localhost:${SERVER_PORT}/wsmaster/api'.
+#    2. Else use the value of CODENVY_API
+#
+# Codenvy Server --> Docker Daemon Progression:
+#    1. Use the value of CODENVY_DOCKER_DAEMON__URL
+#    2. Else, use the value of DOCKER_HOST system variable
+#    3. Else, use Unix socket over unix:///var/run/docker.sock
+#
+# Codenvy Server --> Workspace Connection:
+#    - If CODENVY_DOCKER_SERVER__EVALUATION__STRATEGY is 'default':
```

Comment:
We use server evaluation strategy "codenvy". I'm not sure whether it makes sense to allow user to change implementation unless he knows internals of the product. Also there are other places that influence server -> ws communication. For example our url transformer that converts URLs using patterns provided in property. Also there is behavior influenced by host OS, and whether some property set.
@benoitf probably knows better since he added that code into codebase.